java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -279,6 +287,25 @@ private static Field avroSchemaToField(Schema schema, String name) {
           name = getDefaultFieldName(arrowType);
         }
         return new Field(name, FieldType.nullable(arrowType), Collections.singletonList(structField));
+      case RECORD:
+        List<Field> childFields = new ArrayList<>();
+        for (int i = 0; i < schema.getFields().size(); i++) {
+          final Schema.Field field = schema.getFields().get(i);
+          Schema childSchema = field.schema();
+          childFields.add(avroSchemaToField(childSchema, field.name(), config));
+        }
+        arrowType = new ArrowType.Struct();
+        if (name == null) {
+          name = getDefaultFieldName(arrowType);
+        }
+        return new Field(name, FieldType.nullable(arrowType), childFields);
```

Comment:
Please comment on why you are using FieldType.nullable for the struct type.

Reply:
I think we probably have discussed before, as it's no need to maintain FieldType.nullable with plenty extra logic since vectors could be nullable whatever the flag is.